ngraph/core/src/validation_util.cpp

```diff
@@ -1194,3 +1196,375 @@ void ngraph::evaluate_nodes(std::map<RawNodeOutput, HostTensorPtr>& value_map,
         evaluator.evaluate(value);
     }
 }
+
+bool ngraph::could_propagate(const Output<Node>& output, NodeVector& order)
+{
+    bool status = true;
+
+    std::deque<shared_ptr<Node>> nodes_to_calculate = {output.get_node_shared_ptr()};
```

Comment:
In nG infrastructure we usually operate with raw pointers instead of shared_ptr not to extend node liveness and avoid excess counter increments.
```suggestion
    std::deque<Node *> nodes_to_calculate = {output.get_node()};
```